numba/tests/doc_examples/test_examples.py

```diff
@@ -220,6 +353,189 @@ def g(x, y, res):
+            def init_values(invals, outvals):
+                invals[0] = 6.5
+                outvals[0] = 4.2
+            # magictoken.ex_guvectorize_overwrite.end
+
+            # magictoken.ex_guvectorize_overwrite_call_one.begin
+            invals = np.zeros(shape=(3, 3), dtype=np.float64)
+            # invals == array([[6.5, 6.5, 6.5],
+            #                  [6.5, 6.5, 6.5],
+            #                  [6.5, 6.5, 6.5]])
+
+            outvals = init_values(invals)
+            # outvals == array([[4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2]])
+            # magictoken.ex_guvectorize_overwrite_call_one.end
+
+            self.assertIsInstance(invals, np.ndarray)
+            correct = np.array([
+                [6.5, 6.5, 6.5],
+                [6.5, 6.5, 6.5],
+                [6.5, 6.5, 6.5]])
+            np.testing.assert_array_equal(invals, correct)
+
+            self.assertIsInstance(outvals, np.ndarray)
+            correct = np.array([
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2]])
+            np.testing.assert_array_equal(outvals, correct)
+
+            # magictoken.ex_guvectorize_overwrite_call_two.begin
+            invals = np.zeros(shape=(3, 3), dtype=np.float32)
+            # invals == array([[0., 0., 0.],
+            #                  [0., 0., 0.],
+            #                  [0., 0., 0.]], dtype=float32)
+            outvals = init_values(invals)
+            # outvals == array([[4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2]])
+            # magictoken.ex_guvectorize_overwrite_call_two.end
+
+            self.assertIsInstance(invals, np.ndarray)
+            correct = np.array([
+                [0., 0., 0.],
+                [0., 0., 0.],
+                [0., 0., 0.]], dtype=np.float32)
+            np.testing.assert_array_equal(invals, correct)
+
+            self.assertIsInstance(outvals, np.ndarray)
+            correct = np.array([
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2]])
+            np.testing.assert_array_equal(outvals, correct)
+
+    def test_vectorize_dynamic(self):
+        with captured_stdout():
+            # magictoken.ex_vectorize_dynamic.begin
+            from numba import vectorize
+
+            @vectorize
+            def f(x, y):
+                return x * y
+            # magictoken.ex_vectorize_dynamic.end
+
+            # magictoken.ex_vectorize_dynamic_call_one.begin
+            result = f(3,4)
+            # result == 12
+
+            print(f.types)
+            # ['ll->l']
+            # magictoken.ex_vectorize_dynamic_call_one.end
+
+            self.assertEqual(result, 12)
+            correct = [['ll->l'], ['ll->q']]
+            self.assertIn(f.types, correct)
```

Comment:
Windows test we're failing due to differences in type signatures. On Linux, the output of `f.types == ['ll->l']`, but on Windows 64 the output of `f.types == ['ll->q']`. Do you know what 'q' is? I believe 'l' is for long. I created an issue related to this here #8668. Should I raise it in Numpy instead because there may be other inconsistencies between Windows and Linux?

For now, I'm just testing for both type signatures, but I can change it if needed.

Reply:
`'q'` is a `long long int`. I think this is different on Windows because it has the LLP64 (long long integers and pointers are 64-bit, long integers are 32 bits) data model, whereas 64-bit Linux is LP64 (long integers and pointers are 64 bits) - so it needs to be `q` because it it were `l` on Windows that would be a 32-bit return value. (See the 64-bit Data Models table in https://en.wikipedia.org/wiki/64-bit_computing and the section that it's in).

There are inconsistencies between Windows and Linux, but in this case I believe it to be a platform difference rather than an unexpected issue / bug, so the test will need to check differently on Windows (we have quite a few other Windows-specific deviations in the test suite).